### PR TITLE
Removing jQuery usage from app

### PR DESCRIPTION
**Pull request by dbendaou**
After removing jquery usage from test, this PR aims to remove jquery usage in the app.

TODO : 
- [x] app/components/ember-anchor.js

- [x] app/services/scroll-position-reset.js

- [x] app/mixins/scroll-tracker.js

Related issue https://github.com/ember-learn/ember-api-docs/issues/441

### CODE REVIEW

**Comment by dbendaou** [comment]
Just need to quick refact the `getOffset` method in a util will try to finish it this week

File: app/components/ember-anchor.js [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
 export default AnchorComponent.extend({
-
   // This overrides Ember Anchor to support scrolling within a fixed position element
   _scrollToElemPosition() {
     let qp = this.anchorQueryParam;
     let qpVal = this.get(get(this, 'attrs.a') ? 'a' : `controller.${qp}`);
-    let elem = $(`[data-${qp}="${qpVal}"]`);
-    let offset = elem.offset() ? elem.offset().top : 0;
+    let elem = document.querySelector(`[data-${qp}="${qpVal}"]`);
+    let offset = elem.offsetHeight ? elem.offsetHeight : 0;
```
Comment (Turbo87) [inline app/components/ember-anchor.js]: shouldn't this be `offsetTop`?

**Comment by dbendaou** [comment]
Could anyone review this ? 😅

**Comment by sivakumar-kailasam** [comment]
Sorry about that @dbendaou. I've deployed a preview app to test these changes.

File: app/components/ember-anchor.js [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
     let qp = this.anchorQueryParam;
     let qpVal = this.get(get(this, 'attrs.a') ? 'a' : `controller.${qp}`);
-    let elem = $(`[data-${qp}="${qpVal}"]`);
-    let offset = elem.offset() ? elem.offset().top : 0;
-    if (offset) {
-      $(config.APP.scrollContainerSelector).scrollTop(offset);
+    let elem = document.querySelector(`[data-${qp}="${qpVal}"]`);
+    const offsetTop = elem.offsetHeight;
+
+    if (offsetTop) {
```
Comment (sivakumar-kailasam) [inline app/components/ember-anchor.js]: While testing, I noticed elem was null somewhere (couldn't reproduce it). Could you wrap this block with a check for `elem && elem.offsetHeight`?

File: app/mixins/scroll-tracker.js [hunk 3]
```diff
@@ -15,10 +15,12 @@ export default Mixin.create({
...
       this._super();
       if ((typeof FastBoot === 'undefined') && window.location.search === '?anchor=' ) {
-        let elem = $('#methods');
-        let offset = elem.offset() ? elem.offset().top : 0;
-        if (offset) {
-          $(config.APP.scrollContainerSelector).scrollTop(offset - 10);
+        let elem = document.querySelector('#methods');
+        let offsetTop = elem.offsetHeight;
+
+        if (offsetTop) {
```
Comment (sivakumar-kailasam) [inline app/mixins/scroll-tracker.js]: Same as above.

**Pull request merged**